### ISSUE TITLE
🚀 Feature: "Querie" Database by permissions

### ISSUE THREAD

**Issue by nimos-dev** [appwrite/appwrite#5169]
### 🔖 Feature description

Extend the functionality of: Database List Documents (get a list of all the user's documents in a given collection.) 

Suggestions:

List Documents(X) Get a list of all the user's documents in a given collection where user has permissions X;

List Documents(Read) (Get a list of all the user's documents which the user has read permissions) 
List Documents(Read/Write) (Get a list of all the user's documents which the user has read/write permissions) 
List Documents(Read/Write/Delete) ( Get a list of all the user's documents which the user has read/write/Delete permissions)

### 🎤 Pitch

Today we can List Documents (get a list of all the user's documents in a given collection.) which in practice means all documents to which the user has read rights. It would be nice if you could get all documents based on what permissions the user have. Today you have to implement extra redundant attributes if we want to sort/querry by permission.

Examples of areas of use are content moderation.

Not a critical feature but something that would help avoid boilerplate code and redundant attributes which improves code quality and reduces the risk of errors. Today, when I give a user write rights, I have to both update permissions and then also an attribute field to be able to sort/querry based on whether the user has write permissions or not.

### 👀 Have you spent some time to check if this issue has been raised before?

- [X] I checked and didn't find similar issue

### 🏢 Have you read the Code of Conduct?

- [X] I have read the [Code of Conduct](https://github.com/appwrite/appwrite/blob/HEAD/CODE_OF_CONDUCT.md)

**Comment by dhonig-weizmann** [comment]
Upvoted :)

is there a workaround I can use in the meantime?

**Comment by dhonig-weizmann** [comment]
upvoted :)

Is there any workaround I can use in the meantime?

I tried to use 
`Query.contains('\$permissions', 'read("user:$subjectId")')` but I get an error `Attribute not found in schema: $permissions` 
Using dart_appwrite version 11.0.3

Any help would be most appreciated

**Comment by barshag** [comment]
@dhonig-weizmann  I add user attribute....but i prefere this issue will be push to production soon...(it's just a walkaround..)

**Comment by dhonig-weizmann** [comment]
@barshag, In the end I used the JWT token and read only "my" documents, as I needed all the documents created by a specific client. This worked great :)

**Comment by cosmicsan** [comment]
i upvoted. This features is really really interesting for us.

**Comment by cloudatlas9** [comment]
>  In the end I used the JWT token and read only "my" documents, as I needed all the documents created by a specific client. This worked great :)

@dhonig-weizmann Good point! This is basically the same as querying by permissions, when the goal is to get all documents a user has access to.

**Comment by apperside** [comment]
The thing that may be really useful would be to be able to filter by users in the appwrite dashboard. 
Imagine you get a ticker from a user, it would streamline the problem debugging being able to filter records for which the user has permissions.
